pocs/observatory.py

```diff
@@ -199,43 +198,82 @@ def get_observation(self, *args, **kwargs):
 
         return self.current_observation
 
-    def cleanup_observations(self):
+    def cleanup_observations(self, timeout=600):
         """Cleanup observation list
 
         Loops through the `observed_list` performing cleanup tasks. Resets
         `observed_list` when done
 
+        Args:
+            timeout (int, optional): Timeout for each directory to upload files,
+                defaults to 600 seconds (10 minutes).
```

Comment:
The slowest speed that will work for uploading a full night's worth of data within 24 hours is about 128Kbps (i.e. a modest speed DSL connection). I think that we should use that as our measure of whether it is uploading fast enough. For now, I'd just suggest increasing that default to at least an hour, but later we could use a continuously running process which uploads files that it is informed about over zeromq, and reports progress the same way. That way we're not using Popen.communicate's crude timeout technique.